symphony-ws/src/main/java/se/havochvatten/symphony/service/ReportService.java

```diff
@@ -420,4 +415,290 @@ static void writeComparisonRows(Locale locale, SymphonyBandDto[] meta, int[] fea
+                2 + layersToList.pressures.length,
+                ODF_TITLE_ROWS + layersToList.ecosystems.length);
+            nextCell.setDoubleValue(cmp.cumulativeTotal);
+            nextCell.getOdfElement().setStyleName("totalC");
+
+            // ----- Total sheet -----
+
+            nextCell = totalSheet.getCellByPosition(1, cmpIndex * 7 + ODF_TITLE_ROWS_TOTAL);
+            nextCell.setStringValue(cmp.calculationName);
+            nextCell.getOdfElement().setStyleName("calcName");
+            totalSheet.getRowByIndex(cmpIndex * 7 + ODF_TITLE_ROWS_TOTAL + 2).setHeight(2, false);
+            totalSheet.getRowByIndex(cmpIndex * 7 + ODF_TITLE_ROWS_TOTAL + 5).setHeight(2, false);
+
+            nextCell = totalSheet.getCellByPosition(2, cmpIndex * 7 + ODF_TITLE_ROWS_TOTAL);
+            nextCell.setStringValue("Pressure");
+            nextCell.getOdfElement().setStyleName("pressureHeader");
+            nextCell = totalSheet.getCellByPosition(2, cmpIndex * 7 + ODF_TITLE_ROWS_TOTAL + 1);
+            nextCell.setStringValue("TOTAL");
+            nextCell.getOdfElement().setStyleName("totalHP");
+
+            nextCell = totalSheet.getCellByPosition(2, cmpIndex * 7 + ODF_TITLE_ROWS_TOTAL + 3);
+            nextCell.setStringValue("Ecosystem");
+            nextCell.getOdfElement().setStyleName("pressureHeader");
+            nextCell = totalSheet.getCellByPosition(2, cmpIndex * 7 + ODF_TITLE_ROWS_TOTAL + 4);
+            nextCell.setStringValue("TOTAL");
+            nextCell.getOdfElement().setStyleName("totalHP");
+
+
+            for(p = 0; p < layersToList.pressures.length; p++) {
+                nextCell = totalSheet.getCellByPosition(3 + p, cmpIndex * 7 + ODF_TITLE_ROWS_TOTAL);
+                nextCell.setStringValue(pressureTitles.get(cmp.includedPressures[layersToList.pressures[p]]));
+                nextCell.getOdfElement().setStyleName("pressureHeader");
+                nextCell = totalSheet.getCellByPosition(3 + p, cmpIndex * 7 + ODF_TITLE_ROWS_TOTAL + 1);
+                nextCell.setDoubleValue(cmp.totalPerPressure.get(cmp.includedPressures[layersToList.pressures[p]]));
+                nextCell.getOdfElement().setStyleName("totalP");
+                totalSheet.getColumnByIndex(3 + p).setWidth(ODF_TITLE_COLWIDTH);
+            }
+
+            for(e = 0; e < layersToList.ecosystems.length; e++) {
+                nextCell = totalSheet.getCellByPosition(3 + e, cmpIndex * 7 + ODF_TITLE_ROWS_TOTAL + 3);
+                nextCell.setStringValue(ecoTitles.get(cmp.includedEcosystems[layersToList.ecosystems[e]]));
+                nextCell.getOdfElement().setStyleName("pressureHeader");
+                nextCell = totalSheet.getCellByPosition(3 + e, cmpIndex * 7 + ODF_TITLE_ROWS_TOTAL + 4);
+                nextCell.setDoubleValue(cmp.totalPerEcosystem.get(cmp.includedEcosystems[layersToList.ecosystems[e]]));
+                nextCell.getOdfElement().setStyleName("totalP");
```

Comment:
Are these style names correct?

Reply:
Yes. The named styles are prepared in the template file - I explain it in this comment further up:
https://github.com/havochvatten/MSP-Symphony/blob/6724381e39b52a5444819dd1a04bbafd18d2c628/symphony-ws/src/main/java/se/havochvatten/symphony/service/ReportService.java#L519-L522
Earlier during development, I tediously made all style declarations in the Java code.
I think this is much preferable since the alternative clutters the source unnecessarily.

Reply:
But should it not be "ecoHeader" and "totalE"?

Reply:
Oh I see what you mean! It's correct actually.

This is because there's a difference between how the "total" sheet and the individual sheets sums are rendered. On the individual sheets we use "totalE" for ecosystem totals (horizontal sum). But on the total sheet we "reuse" the format for the vertical sum "pressureE", and the column header format "pressureHeader" for both component types.

We could rename them thus, I guess: _columnHeader / totalV / totalH_ etc. . Still, they're descriptive for the detailed sheet..